### PR TITLE
Button update

### PR DESCRIPTION
**Pull request by eldiegod**
Fixes: #2463

- IconButton component got removed; use the Button component with icon prop and no children.
- iconOnly prop got removed; for a while, this prop did nothing as we calculate this when an icon is passed, and no children are passed.
- I removed the ButtonBase component as it served no purpose.
- removed unused or duplicated styles.
- updated the text only variants

I spent hours trying to find a way around this just with types. Still, due to the text only variants being different from the default variants, in an attempt to properly handle this, I renamed the `textOnly` to `_textOnly` to be used internally only.
I also created the TextButton component, the same as our Button component but with correct typings for text variants.

### CODE REVIEW

File: packages/atlas/src/components/Banner/Banner.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/Banner/Banner.styles.ts]: I think we should use absolute import path here

File: packages/atlas/src/components/Carousel/Carousel.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/Carousel/Carousel.styles.ts]: I think we should use absolute import path here

File: packages/atlas/src/components/DrawerHeader/DrawerHeader.tsx
Comment (kdembler) [inline packages/atlas/src/components/DrawerHeader/DrawerHeader.tsx]: I think we should use absolute import path here

File: packages/atlas/src/components/ProgressDrawer/ProgressDrawer.tsx
Comment (kdembler) [inline packages/atlas/src/components/ProgressDrawer/ProgressDrawer.tsx]: I think we should use absolute import path here

File: packages/atlas/src/components/Searchbar/Searchbar.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/Searchbar/Searchbar.styles.ts]: I think we should use absolute import path here

File: packages/atlas/src/components/Searchbar/Searchbar.tsx
Comment (kdembler) [inline packages/atlas/src/components/Searchbar/Searchbar.tsx]: I think we should use absolute import path here

File: packages/atlas/src/components/Step/Step.tsx
Comment (kdembler) [inline packages/atlas/src/components/Step/Step.tsx]: I think we should use absolute import path here

File: packages/atlas/src/components/_buttons/Button/Button.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/_buttons/Button/Button.styles.ts]: I don't think this object is needed at all?

File: packages/atlas/src/components/_buttons/Button/Button.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/_buttons/Button/Button.styles.ts]: We agreed to not use this type of "style in a string" as it bypasses stylelint: https://github.com/Joystream/atlas/blob/master/docs/styleguide.md#conditional-styles

File: packages/atlas/src/components/_buttons/Button/Button.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/_buttons/Button/Button.styles.ts]: This shouldn't be Capitalized since it's not a component

File: packages/atlas/src/components/_buttons/Button/Button.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/_buttons/Button/Button.styles.ts]: Same here, shouldn't be capitalized

File: packages/atlas/src/components/_buttons/HamburgerButton/HamburgerButton.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/_buttons/HamburgerButton/HamburgerButton.styles.ts]: Should use absolute path

File: packages/atlas/src/components/_inputs/FileSelect/FileSelect.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/_inputs/FileSelect/FileSelect.styles.ts]: This component seems to not be used anywhere, let's remove it

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts
Comment (kdembler) [inline packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts]: Should this be a button actually versus just a regular link?

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx
Comment (kdembler) [inline packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx]: You removed the `you` functionality here

File: packages/atlas/src/components/_overlays/Dialog/Dialog.tsx
Comment (kdembler) [inline packages/atlas/src/components/_overlays/Dialog/Dialog.tsx]: Let's destruct `primaryButton` at the end so that consumer is able to override if needed and so we're consistent with the secondary button

File: packages/atlas/src/utils/types.d.ts
Comment (kdembler) [inline packages/atlas/src/utils/types.d.ts]: This should go into `src/types`

**Review by kdembler — CHANGES_REQUESTED** [review]
Good job in general, left some mostly cosmetic comments. From the visual standpoint, I've noticed that all the text buttons have the same height (24px) regardless of the size:
Screenshot 2022-05-10 at 14 50 09

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx
Comment (eldiegod) [reply packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx]: ah fak

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts
Comment (eldiegod) [reply packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts]: hmm i'm not sure rn but I think it was a button in design

**Comment by eldiegod** [comment]
> ons have the same height (24px) regardless of the size:

@kdembler yeah, it looks like that but it's because of the flex parent component.

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts
Comment (kdembler) [reply packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts]: I don't think it is, at least in some cases, just a text here:
Screenshot 2022-05-10 at 22 15 10

Comment (toiletgranny) [reply packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts]: I can confirm it is just a text in here. If it were a button, it would be marked as such in the layers panel.

**Review by toiletgranny — CHANGES_REQUESTED** [review]
Overall, it's very nice! Here are the things I found:
1. When hovering over any button now, I can feel severe stuttering in the transition animation, no matter which CSS property is being animated. At the same time, in the "Actions" tab on Storybook, I can see quite a lot of onMouseMove events being recorded every time my cursor moves, even by 1px. Could it be connected?
2. The height of the last row of Text Only buttons (Tertiary) is off by 1px in all three sizes, eg. instead of `24px`, it's `23px`. I can see you apply a negative margin vertically on both sides (`margin-top: -0.5px; margin-bottom: -0.5px;`), and I assume it's to accommodate for the 1px bottom border in all other Text Only buttons, which brings me to my next point, which is that...
3. In my opinion, the current specification for CSS borders is just wrong. As far as I know, the `box-sizing: border-box` property would only do its job if you specifically set the element's width and height, which rarely ever happens in the context of UI components such as buttons. Therefore, in Figma, I used absolute positioning for rendering bottom borders. Perhaps you'll find this approach (or something else, eg. `:before`/ `:after` pseudo-elements) more suitable in this case? I'm leaving this as a loose suggestion.

**Comment by drillprop** [comment]
> When hovering over any button now, I can feel severe stuttering in the transition animation, no matter which CSS property is being animated. At the same time, in the "Actions" tab on Storybook, I can see quite a lot of onMouseMove events being recorded every time my cursor moves, even by 1px. Could it be connected?

@eldiegod I used `onMouseMove` and `onMouseOut` in the ChannelCard(check the FollowButton), but I made a mistake with selecting the right handler. We should use `onMouseEnter` and `onMouseLeave` here instead, which is much more performant(it's triggered basically only once when you hover and unhover on the button). Could you replace it and remove `onMouseMove` and `onMouseOut` props?

**Comment by eldiegod** [comment]
> 3\. context of UI components such as buttons. Therefore, in Figma, I used absolute positioning for rendering bottom borders. Perhaps you'll find this approach (or something else, eg. `:before`/ `:after` pseudo-elements) more suitable in this case? I'm leaving this as a loose suggestion.

@toiletgranny I think the issues should be solved now, point 2 was a really easy fix so I didn't wanna go about changing much of the border now but I agree with you nowadays I build this type of thing using :before/ :after

**Comment by eldiegod** [comment]
> > When hovering over any button now, I can feel severe stuttering in the transition animation, no matter which CSS property is being animated. At the same time, in the "Actions" tab on Storybook, I can see quite a lot of onMouseMove events being recorded every time my cursor moves, even by 1px. Could it be connected?
> 
> @eldiegod I used `onMouseMove` and `onMouseOut` in the ChannelCard(check the FollowButton), but I made a mistake with selecting the right handler. We should use `onMouseEnter` and `onMouseLeave` here instead, which is much more performant(it's triggered basically only once when you hover and unhover on the button). Could you replace it and remove `onMouseMove` and `onMouseOut` props?

@drillprop yep fixed 👍

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts
Comment (eldiegod) [reply packages/atlas/src/components/_nft/NftWidget/NftWidget.styles.ts]: replaced it with a normal Link + Text

**Review by kdembler — APPROVED** [review]
The hamburger button is no longer rounded:
Screenshot 2022-05-12 at 12 42 00

Haven't found any other issues though, LGTM once 👆 if fixed, let's get an approval from Adam as well

**Review by toiletgranny — APPROVED** [review]

**Pull request merged**
